### PR TITLE
snap: provide a useful error message if gdbserver is not installed

### PR DESCRIPTION
**Pull request by mvo5**
There was some confusion recently where "gdbserver" should be
installed when using "snap run --gdbserver". There is no early
check and useful error message. This commit fixes this.

Based on https://github.com/snapcore/snapd/pull/9908

### CODE REVIEW

File: cmd/snap/cmd_run.go [hunk 1]
```diff
@@ -1175,6 +1176,12 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook stri
 	} else if x.Gdb {
 		return x.runCmdUnderGdb(cmd, envForExec)
 	} else if x.useGdbserver() {
+		if _, err := exec.LookPath("gdbserver"); err != nil {
+			if xerrors.Is(err, exec.ErrNotFound) {
+				return fmt.Errorf("please install gdbserver on your system")
```
Comment (bboozzoo) [inline cmd/snap/cmd_run.go:1181]: We should have a similar check for `gdb` too AFAICT.

**Review by bboozzoo — APPROVED** [review]
last commit  LGTM

**Review by pedronis — APPROVED** [review]
last commit looks alright

**Review by anonymouse64 — APPROVED** [review]
LGTM

**Pull request merged**